### PR TITLE
[Merged by Bors] - make TextLayoutInfo a Component

### PR DESCRIPTION
**Pull request by yrns**
# Objective

Make `TextLayoutInfo` more accessible as a component, rather than internal to `TextPipeline`. I am working on a plugin that manipulates these and there is no (mutable) access to them right now.

## Solution

This changes `TextPipeline::queue_text` to return `TextLayoutInfo`'s rather than storing them in a map internally. `text2d_system` and `text_system` now take the returned `TextLayoutInfo` and store it as a component of the entity. I considered adding an accessor to `TextPipeline` (e.g. `get_glyphs_mut`) but this seems like it might be a little faster, and also has the added benefit of cleaning itself up when entities are removed. Right now nothing is ever removed from the glyphs map.

## Changelog

Removed `DefaultTextPipeline`. `TextPipeline` no longer has a generic key type. `TextPipeline::queue_text` returns `TextLayoutInfo` directly.

## Migration Guide

This might break a third-party crate? I could restore the orginal TextPipeline API as a wrapper around what's in this PR.

### CODE REVIEW

File: crates/bevy_text/src/pipeline.rs [hunk 1]
```diff
@@ -14,28 +13,19 @@ use crate::{
...
-        TextPipeline {
-            brush: GlyphBrush::default(),
-            glyph_map: Default::default(),
-            map_font_id: Default::default(),
-        }
-    }
-}
-
+#[derive(Component, Clone, Default, Debug)]
 pub struct TextLayoutInfo {
```
Comment (alice-i-cecile) [inline crates/bevy_text/src/pipeline.rs:23]: This will be much more visible; can you add some docs?

Comment (yrns) [reply crates/bevy_text/src/pipeline.rs:23]: Yes!

File: crates/bevy_text/src/pipeline.rs [hunk 2]
```diff
@@ -14,28 +13,21 @@ use crate::{
...
-            brush: GlyphBrush::default(),
-            glyph_map: Default::default(),
-            map_font_id: Default::default(),
-        }
-    }
-}
-
+/// Render information (scaled glyphs and size) for a corresponding
+/// [`Text`](crate::Text) component. Generated via [`TextPipeline::queue_text`].
+#[derive(Component, Clone, Default, Debug)]
```
Comment (Weibye) [inline crates/bevy_text/src/pipeline.rs:24]:
```suggestion
/// Render information for a corresponding [`Text`](crate::Text) component.
///
///  Contains scaled glyphs and their size. Generated via [`TextPipeline::queue_text`].
#[derive(Component, Clone, Default, Debug)]
```

I'm not sure it reads better?

**Review by Weibye — COMMENTED** [review]
I really like the change, the code looks good as far as I can tell :) Tiny comment before approving:

**Review by alice-i-cecile — APPROVED** [review]

**Comment by yrns** [comment]
Brought this up-to-date with the main branch. I wish github displayed the diffs in such a way that it was clearer most of the changed lines are just indentation changes due to removing the surrounding `if let` conditionals.

**Review by Weibye — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
More merge conflicts :( Sorry @yrns; I was on vacation. If you fix those up for us I'll merge it ASAP.

**Comment by yrns** [comment]
Merged and tested the ui example.

**Comment by alice-i-cecile** [comment]
Thanks!

bors r+

**Pull request closed without merging**